First commits

felddy opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

This is a porting of the [`skeleton` Python script](https://github.com/cisagov/development-guide/blob/develop/project_setup/scripts/skeleton) from our development guide to a [`gh` CLI extension](https://github.blog/2021-08-24-github-cli-2-0-includes-extensions/).

## 💭 Motivation and context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

This will make it even easier for developers to get a new skeletonized project off the ground.  

Steps:
 - [`brew install gh`](https://cli.github.com) (or your environment's equivalent) 
 - `gh extension install cisagov/gh-skeleton` 
 - `gh skeleton list`
 - `gh skeleton clone skeleton-generic my-new-project`

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->

<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

I've tested this by installing the extension locally. 
I probably look like a crazy man with the number of times I've created the `felddy/test` repo and then quickly nuked it.
```console
gh skeleton clone --dest-org felddy skeleton-generic test
```


## 📷 Screenshots (if appropriate) ##

<!-- Remove this section and header if not needed -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

felddy committed:
Add ported skeleton script

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,190 @@
+  git remote add origin git@github.com:"${org_name}/${new_repo}".git
+
+  log_info "Search and replace repository name in source files."
+  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
```
Rather than the overhead of regex. Unless you're anticipating a `.git` directory somewhere else in the path.

```suggestion
  find . -path ./.git -prune -false -o -type f -exec \
```

felddy replied:
Good suggestions.  I'm going to take them all except the regex replacement.  I don't expect we'll have skeletons with embedded `.git` directories.  This regex has served us well, and... if it ain't broke... ;)

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,190 @@
+  git add --verbose .
+
+  log_info "Commit staged files to the new repository."
+  git commit --message "Rename repository references after clone."
```
[Do not end the subject line with a period](https://chris.beams.io/posts/git-commit/#end)

```suggestion
  git commit --message "Rename repository references after clone"
```

felddy committed:
Improve handling of change dir and option defaults

mcdonnnj commented:
@felddy You should be fine to test it live by changing to the cache directory where it is cloned and issuing a `git switch first-commits`.

```console
$ gh extension install cisagov/gh-skeleton
Cloning into '<home directory>/.local/share/gh/extensions/gh-skeleton'...
remote: Enumerating objects: 645, done.
remote: Counting objects: 100% (645/645), done.
remote: Compressing objects: 100% (280/280), done.
remote: Total 645 (delta 318), reused 645 (delta 318), pack-reused 0
Receiving objects: 100% (645/645), 193.15 KiB | 5.37 MiB/s, done.
Resolving deltas: 100% (318/318), done.
$ cd ~/.local/share/gh/extensions/gh-skeleton/
$ git switch first-commits
Branch 'first-commits' set up to track remote branch 'first-commits' from 'origin'.
Switched to a new branch 'first-commits'
$ gh skeleton --help
Helper tool to start a new github project from a skeleton github repository.

Usage:
  gh skeleton (-h | --help)
  gh skeleton list [--org=<name>]
  gh skeleton clone [options] <parent-repo-name> <new-repo-name>

Options:
  -c --change-dir=<dir>  Create clone in this directory.
  -h --help              Show this message.
  -o --org=<name>        Organization to search [default: cisagov].
```

felddy committed:
Update documentation with extension information

felddy committed:
Improve project summary and add link to skeletons

felddy committed:
Make instructions less wordy

felddy committed:
Add output for creating first-commits PR

felddy committed:
Change return code for missing clone arguments

felddy committed:
Normalize error return codes

dav3r commented on README.md:
```diff
@@ -2,22 +2,57 @@
-well as [pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This extension for the [`gh` CLI] provides the ability easily start new projects
```
Missing word:
```suggestion
This extension for the [`gh` CLI] provides the ability to easily start new projects
```

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,193 @@
+
+USAGE=$(
+  cat << END_OF_LINE
+gh extension to start a new github project from a skeleton github repository.
```
We always strive to use proper capitalization:
```suggestion
gh extension to start a new GitHub project from a skeleton GitHub repository.
```

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,193 @@
+  "
+
+  # shellcheck disable=2016
+  # disable single quote expansion warning
```
I'd prefer to see a bit more explanation about why you are disabling this warning.

felddy replied:
See: 6c2f104

dav3r reviewed (changes requested):
I have a few small changes to request, but this is some hot sauce! 🔥 🌶️

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,193 @@
+
+# Determine if an executable is in the PATH
+if ! type -p perl > /dev/null; then
+  log_error "perl not found on the system and is required" >&2
```
Should the redirect to `stderr` be part of the `log_error()` function?

mcdonnnj replied:
Capitalization.

```suggestion
  log_error "Perl not found on the system and is required" >&2
```

mcdonnnj reviewed (changes requested):
This should be an excellent alternative to the `skeleton` Python script. Just some small changes for your consideration.

felddy committed:
Remove errant period from commit message

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Add missing word to extension summary

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

felddy committed:
Properly capitalize GitHub

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

felddy committed:
Properly capitalize Perl

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Move redirect to stderr into error log function

felddy committed:
Elaborate shellcheck disable comment

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,193 @@
+  "
+
+  # shellcheck disable=2016
+  # Disable single quote expansion warning as we want to supress parameter expansion in a Go template.
```
Fix a typo.  Also, this comment line is pretty long - why not split it up?
```suggestion
  # Disable single quote expansion warning as we want to suppress
  # parameter expansion in a Go template.
```

dav3r reviewed (approved):
☠️ 🌶️ 

Note my one small additional comment suggestion.

felddy committed:
Correct typo and rewrap comment

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

felddy commented on gh-skeleton:
```diff
@@ -0,0 +1,194 @@
+    cd ${CHANGE_DIR:+${CHANGE_DIR}/}${new_repo}
+    git checkout -b first-commits
+    git push origin develop first-commits --set-upstream
+    gh pr create --title "First commits" --assignee=@me --web
```
@cisagov/team-ois  I was hoping to get some feedback on this jazz, and wanted to make sure you didn't overlook it.

This is to reinforce our guidance in the [project setup](https://github.com/cisagov/development-guide/tree/develop/project_setup#create-an-initial-pull-request) document.   

I was actually thinking about having the script attempt to do these steps itself, but there is a good chance that the remote repo wouldn't yet exist, and the use doesn't have permissions to do a `gh repo create --public cisagov/my-new-repo`

Note that in order to use our PR templates the PR creation needed to be done using the `--web` flag.  Try it out to see how this works. 

Any thoughts?

dav3r replied:
How about if the script checks if `cisagov/my-new-repo` exists already.  If it doesn't, display a message like: "Please create `cisagov/my-new-repo` at github.com, then press Enter to continue."

I agree that it would be slick to have the script create the branch and initial PR here.

jsf9k replied:
I would possibly run the script authenticated, so it would be great if it could _try_ to create the repo and only ask the user to create it if that fails.

felddy replied:
@jsf9k I think that is a good idea. `gh` does have a concept of an authenticated user.  I think this can work.

mcdonnnj replied:
If this is creating a `first-commits` PR in what appears to be an initial state then I think the `--draft` flag should be used.

felddy replied:
I had the same idea about using `--draft`.  Unfortunately `gh` doesn't allow this.

```console
❱ gh pr create --title "First commits" --assignee=@me --draft --web
the `--draft` flag is not supported with `--web`
```

felddy replied:
Automation added in 667c7cd.

felddy committed:
Document requirement that user is logged in

felddy committed:
Add automation for as many steps as possible

- Requires an authenticated gh user to use clone sub-command.
- Add ability to specify destination organization.
- Attemps to create the remote repository.
- Handles remote repository already existing.
- Outputs manual steps for user to folllow if remote creation fails.
- Handles gh cli adding its own origins.
- Commits lineage.yml to develop branch.
- Commits search and replace to first-commits so first PR isn't empty.

felddy commented:
w.r.t. 667c7cd 
Probably should have saved this extra automation version 2 of the extension.
I'm going to request re-reviews since a lot has changed.
No rush.  Have a great weekend.

felddy committed:
Remove equals signs from long flag usage

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  log_info "Disable pushing to the upstream (parent) repository."
+  git remote set-url --push "${src_repo}" no_push
+
+  log_info "Create the lineage.yml file."
```
Info messages should all use the same tense:
```suggestion
  log_info "Creating ${dest_org}/${dest_repo} from skeleton ${src_org}/${src_repo}."

  log_info "Cloning skeleton remote repository to the new local repository."
  git clone --origin "${src_repo}" git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"

  cd "${dest_repo}"

  log_info "Disabling pushing to the upstream (parent) repository."
  git remote set-url --push "${src_repo}" no_push

  log_info "Creating the lineage.yml file."
```

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  log_info "Stage modified files."
+  git add --verbose .
+
+  log_info "Commit staged files to the first-commits branch."
```
More verb tense agreement in log messages:
```suggestion
  log_info "Staging modified files."
  git add --verbose .

  log_info "Committing staged files to the develop branch."
  git commit --message "Adding lineage configuration"

  log_info "Creating first-commits branch."
  git checkout -b first-commits

  log_info "Searching and replacing repository name in source files."
  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
    perl -pi -e "s/${src_repo}/${dest_repo}/g" {} \;

  log_info "Staging modified files."
  git add --verbose .

  log_info "Committing staged files to the first-commits branch."
```

felddy replied:
One of the previous suggestions has now angered this one, preventing me from committing it.   I'll see if I can resolve this differently.

felddy replied:
Thank you, I've incorporated your suggestions.  See:  66cf84d

dav3r commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+      log_ok "The remote repository ${dest_org}/${dest_repo} was successfully created."
+      ;;&
+    existed | failed)
+      log_info "Add a new remote origin for the this repository."
```
Another one:
```suggestion
      log_info "Adding a new remote origin for the this repository."
```

felddy replied:
This is a log message.  Not a commit message.

dav3r reviewed (changes requested):
I'd like to see some log messages changed for consistency.

jsf9k commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  git add --verbose .
+
+  log_info "Commit staged files to the develop branch."
+  git commit --message "Adding lineage configuration"
```
Commit messages should be in the imperative mood.
```suggestion
  git commit --message "Add lineage configuration"
```

felddy committed:
Change commit message to imperative mood

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

felddy committed:
Change tense of logging messages to be consistent

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
🔥 🌶️

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+
+# Determine if an executable is in the PATH
+if ! type -p perl > /dev/null; then
+  log_error "Perl not found on the system and is required"
```
Just for consistency.

```suggestion
  log_error "Perl not found on the system and is required."
```

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+
+USAGE=$(
+  cat << END_OF_LINE
+gh extension to start a new GitHub project from a skeleton GitHub repository.
```
I think it's better to be verbose in the tool's name instead of just using `gh`.

```suggestion
GitHub CLI extension to start a new GitHub project from a skeleton GitHub repository.
```

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  log_info "Disabling pushing to the upstream (parent) repository."
+  git remote set-url --push "${src_repo}" no_push
+
+  log_info "Creating the lineage.yml file."
+  cat << END_OF_LINE > .github/lineage.yml
+---
+lineage:
+  skeleton:
+    remote-url: https://github.com/${src_org}/${src_repo}.git
+version: '1'
+END_OF_LINE
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the develop branch."
+  git commit --message "Add lineage configuration"
+
+  log_info "Creating first-commits branch."
+  git checkout -b first-commits
+
+  log_info "Searching and replacing repository name in source files."
+  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
+    perl -pi -e "s/${src_repo}/${dest_repo}/g" {} \;
+
+  log_info "Staging modified files."
+  git add --verbose .
+
+  log_info "Committing staged files to the first-commits branch."
+  git commit --message "Rename repository references after clone"
```
We have traditionally updated references as the first commit to the `develop` branch post-clone (this included the [cisagov/action-lineage](https://github.com/cisagov/action-lineage) configuration) and then the `first-commits` branch is initial functionality. If we want to change away from that I'm ok with it, but I think the [cisagov/action-lineage](https://github.com/cisagov/action-lineage) configuration should be done in `first-commits` as well in that case.

I'm of the mind that either the initial clone is unchanged and the `first-commits` branch includes all initial changes or we do all housekeeping as an initial set of commits to the `develop` branch before beginning the `first-commits` branch.

```suggestion
  log_info "Searching and replacing repository name in source files."
  find . \( ! -regex '.*/\.git/.*' \) -type f -exec \
    perl -pi -e "s/${src_repo}/${dest_repo}/g" {} \;

  log_info "Staging modified files."
  git add --verbose .

  log_info "Committing staged files to the develop branch."
  git commit --message "Rename repository references after clone"

  log_info "Creating the lineage.yml file."
  cat << END_OF_LINE > .github/lineage.yml
---
lineage:
  skeleton:
    remote-url: https://github.com/${src_org}/${src_repo}.git
version: '1'
END_OF_LINE

  log_info "Staging modified files."
  git add --verbose .

  log_info "Committing staged files to the develop branch."
  git commit --message "Add lineage configuration"

  log_info "Creating first-commits branch."
  git checkout -b first-commits
```

jsf9k replied:
I like the latter (all `clone`-related housekeeping is pushed to `develop`, then create the `first-commits` branch from that) since that's the way we've been doing it and we haven't had any issues.

felddy replied:
Now that I've gone back and tested it... I realize there was a reason I was putting the lineage configuration into `develop` and the search and replace changes into the `first-commits` branch:

This gives `first-commits` a commit, which allows the PR that is opened at the end of the script to succeed.  The way it is now, `first-commits` has no commits, and the PR creation fails as a noop.

felddy replied:
The empty PR will suffice for now.  Most people will understand what is going on when they see the resulting PR page.  

@mcdonnnj Let's get this resolved so I can move on with #3.

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  remote_repo_name=$(gh repo view --json name --jq ".name" "${dest_org}/${dest_repo}") || :
+  if [[ -n ${remote_repo_name} ]]; then
+    log_ok "${dest_org}/${dest_repo} exists."
+    remote_repo_status="existed"
```
Thoughts on switching to `exists` from `existed`? The tense sounds off to me.

jsf9k replied:
I think it is correct as it stands.  Changing to "existed" would be _au rebours_, in my opinion.

mcdonnnj replied:
I'm confused by your statement. I was suggesting:

```suggestion
    remote_repo_status="exists"
```

jsf9k replied:
Then that would be switching from "existed" to "exists", not the other way around.  Given the transposition error I thought you were referring to the log comment in line 146.

Please edit your original comment if I'm understanding correctly.  With the transposition error it is highly misleading.

jsf9k replied:
Ah, I keep misreading "to `exists` from `existed`" as "from `existed` to `exists`".  Doh!

Disregard me!

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+      log_ok "The remote repository ${dest_org}/${dest_repo} was successfully created."
+      ;;&
+    existed | failed)
+      log_info "Adding a new remote origin for the this repository."
```
```suggestion
      log_info "Adding a new remote origin for the repository."
```

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  log_info "Creating ${dest_org}/${dest_repo} from skeleton ${src_org}/${src_repo}."
+
+  log_info "Cloning skeleton remote repository to the new local repository."
+  git clone --origin "${src_repo}" git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"
```
This creates a configuration that mirrors what our other projects around skeleton repositories use.

```suggestion
  git clone --origin skeleton git@github.com:"${src_org}/${src_repo}".git "${dest_repo}"
```

jsf9k replied:
Are you sure @mcdonnnj?  All my remotes are named `skeleton-docker`, `skeleton-ansible-role`, etc. (and that's how I like it).

mcdonnnj replied:
I was referring to things like the [cisagov/action-lineage conflict PR description](https://github.com/cisagov/action-lineage/pull/30).

Relevant portion:

```console
git clone git@github.com:cisagov/action-lineage.git action-lineage
cd action-lineage
git remote add skeleton https://github.com/cisagov/skeleton-python-library.git
git remote set-url --push skeleton no_push
git switch develop
git checkout -b lineage/skeleton --track origin/develop
git pull skeleton HEAD
git checkout --ours -- .github/lineage.yml
git add .github/lineage.yml
git status
```

mcdonnnj replied:
I see that the [`skeleton` script](https://github.com/cisagov/development-guide/blob/d15d4445ee56610d6d4059192ee984aee9f037e2/project_setup/scripts/skeleton#L59-L62) mirrors what you're talking about. In that case we need to get consistent language in place everywhere we're directing things about setting up repository specifics with `git`.

jsf9k replied:
Again, `git remote add skeleton` is incorrect in my opinion.  I think we want the `${src_repo}` there as the name of the skeleton remote because it provides useful information.  If this is inconsistent with instructions given elsewhere then I suggest changing the other instructions.

felddy replied:
I agree.  Let's find the documentation that is in conflict, and update it.

mcdonnnj replied:
I've created https://github.com/cisagov/action-lineage/issues/36 to track fixing this in that repository. We probably want to consider getting this into [cisagov/development-guide](https://github.com/cisagov/development-guide) at some point to solidify the expectation.

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+  cd "${dest_repo}"
+
+  log_info "Disabling pushing to the upstream (parent) repository."
+  git remote set-url --push "${src_repo}" no_push
```
This just mirrors the change I suggested above.

```suggestion
  git remote set-url --push skeleton no_push
```

jsf9k replied:
See my previous comments on this same issue.

felddy replied:
See my previous comment agreeing with the @jsf9k's previous comment.

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+      git push origin develop first-commits --set-upstream
+      log_info "Opening a new pull request for the first-commits branch."
+      log_info "If prompted, please select ${dest_org}/${dest_repo} as the base repository."
+      gh pr create --title "First commits" --assignee=@me --web
```
I believe this should be a draft if we're creating it before any work has been done.

```suggestion
      gh pr create --title "First commits" --assignee=@me --draft --web
```

jsf9k replied:
Ah yes, that is excellent.

felddy replied:
This has been covered previously: https://github.com/cisagov/gh-skeleton/pull/1#discussion_r697752596

![gandalf confused](https://user-images.githubusercontent.com/3229435/131731315-ddf899dd-036f-4b34-b381-4d544ada2de2.gif)

mcdonnnj replied:
I missed when this was shown as impossible in https://github.com/cisagov/gh-skeleton/pull/1#discussion_r697752596, oops!

mcdonnnj commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+the local repository to the remote and create the initial pull request:
+    cd ${CHANGE_DIR:+${CHANGE_DIR}/}${dest_repo}
+    git push origin develop first-commits --set-upstream
+    gh pr create --title "First commits" --assignee=@me --web
```
Same suggestion about draft PRs.

```suggestion
    gh pr create --title "First commits" --assignee=@me --draft --web
```

mcdonnnj reviewed (changes requested):
More feedback for your consideration. Thanks for the continued work on this extension!

felddy committed:
Add period to log error message

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Add verbose tool name to usage string

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Remove extra word from log message

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Move clone-related housekeeping to develop branch

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy committed:
Normalize tense of flag to match log statements

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

felddy commented on gh-skeleton:
```diff
@@ -0,0 +1,254 @@
+
+  # Handle the different outcomes of the remote repository.
+  case $remote_repo_status in
+    created)
+      log_ok "The remote repository ${dest_org}/${dest_repo} was successfully created."
+      ;;&
+    existed | failed)
+      log_info "Adding a new remote origin for the repository."
+      # We can add the remote origin now since gh was unable to do it.
+      git remote add origin git@github.com:"${dest_org}/${dest_repo}".git
+      ;;&
+    created | existed)
+      log_info "Pushing develop and first-commit branches to the remote."
+      git push origin develop first-commits --set-upstream
+      log_info "Opening a new pull request for the first-commits branch."
+      log_info "If prompted, please select ${dest_org}/${dest_repo} as the base repository."
+      gh pr create --title "First commits" --assignee=@me --web
+      log_ok "Success!"
+      ;;
+    failed)
+      log_warn "The remote repository ${dest_org}/${dest_repo} counld not be created."
+      cat << END_OF_LINE
```
Change tense of flag to match the previous changes

```suggestion
    created)
      log_ok "The remote repository ${dest_org}/${dest_repo} was successfully created."
      ;;&
    exists | failed)
      log_info "Adding a new remote origin for the repository."
      # We can add the remote origin now since gh was unable to do it.
      git remote add origin git@github.com:"${dest_org}/${dest_repo}".git
      ;;&
    created | exists)
      log_info "Pushing develop and first-commit branches to the remote."
      git push origin develop first-commits --set-upstream
      log_info "Opening a new pull request for the first-commits branch."
      log_info "If prompted, please select ${dest_org}/${dest_repo} as the base repository."
      gh pr create --title "First commits" --assignee=@me --web
      log_ok "Success!"
      ;;
    failed)
      log_warn "The remote repository ${dest_org}/${dest_repo} counld not be created."
      cat << END_OF_LINE
```

felddy committed:
Change tense of flag to match the previous changes

mcdonnnj reviewed (approved):
Works for me. Looking forward to future enhancements 🚀🌕

felddy committed:
Merge branch 'develop' into first-commits
